File: day11/maxDigit.cpp
#include <iostream>
#include <string>
#include <cctype>

int main() {
    std::string input;
    
    while (true) {
        std::cout << "Enter a number and I will extract the maximum digit: ";
        std::cin >> input;

        bool is_valid = true;
        for (char c : input) {
            if (!isdigit(c)) {
                is_valid = false;
                break;
            }
        }

        if (is_valid) {
            char max_digit = '0';
            for (char c : input) {
                if (c > max_digit) {
                    max_digit = c;
                }
            }
            std::cout << "The maximum digit is: " << max_digit << "\n";
            break;
        } else {
            std::cout << "Please enter only digits.\n";
        }
    }

    return 0;
}

File: day11/MaxDigit.java
import java.util.Scanner;

public class MaxDigit {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.print("Enter a number and I will extract the maximum digit: ");
            String input = scanner.nextLine();

            try {
                // Try to parse to ensure input is numeric
                Integer.parseInt(input);

                int maxDigit = 0;
                for (char c : input.toCharArray()) {
                    int digit = Character.getNumericValue(c);
                    if (digit > maxDigit) {
                        maxDigit = digit;
                    }
                }

                System.out.println("The maximum digit is: " + maxDigit);
                break;
            } catch (NumberFormatException e) {
                System.out.println("Please enter only digits.\n");
            }
        }

        scanner.close();
    }
}
